src/jsx/components/input_info.jsx

```diff
@@ -4,32 +4,46 @@ var pd = require('pretty-data').pd;
 import { saveAs } from "file-saver";
 import { Modal, Button, Glyphicon } from 'react-bootstrap';
 import ReactJson from 'react-json-view';
+import Alignment from 'alignment.js';
+
+const d3 = require('d3');
+
 
 class InputInfo extends React.Component {
   constructor(props){
     super(props);
-    this.state = { showModal: false };
+    this.state = { 
+      showModal: false,
+      fasta: ''
+    };
   }
   close() {
-    this.setState({ showModal: false});
+    this.setState({ showModal: false });
   }
-  open() {
-    this.setState({ showModal: true});
+  open(content) {
+    this.setState({ showModal: content });
   }
   saveTheJSON() {
     var blob = new Blob([pd.json(this.props.json)], {
       type: "text/json:charset=utf-8;"
     });
     saveAs(blob, "result.json");
   }
+  componentDidMount(){
+    const self = this;
+    d3.json(window.location.href+"/fasta", (err, data) => {
+      self.setState({fasta: data.fasta});
+    });
+  }
   render(){
     if (!this.props.input_data) return <div />;
     var is_full_path = this.props.input_data["file name"].indexOf("/") != -1,
       filename = is_full_path
         ? _.last(this.props.input_data["file name"].split("/"))
         : this.props.input_data["file name"],
       on_datamonkey = !this.props.hyphy_vision,
-      show_partition_button = on_datamonkey && this.props.gard;
+      show_partition_button = on_datamonkey && this.props.gard, 
+      fasta = this.state.fasta;
```

Comment:
You may want to just use `this.state.fasta` below rather than define it as a variable here. Either way is ok but because it's only used once, defining it as a variable seems unnecessary.